common/src/currency.rs

```diff
@@ -1,48 +1,131 @@
-use sp_runtime::RuntimeDebug;
+use sp_std::convert::From;
+use sp_std::convert::Into;
+use sp_std::convert::TryFrom;
+use sp_std::convert::TryInto;
+use sp_std::fmt;
+use sp_std::str;
+
+use stellar::types::AssetAlphaNum12;
+use stellar::types::AssetAlphaNum4;
+use stellar::PublicKey;
+use substrate_stellar_sdk as stellar;
 
 use codec::{Decode, Encode};
 
+pub type Bytes4 = [u8; 4];
+pub type Bytes12 = [u8; 12];
+pub type AssetIssuer = [u8; 32];
+
 #[cfg(feature = "std")]
 use serde::{Deserialize, Serialize};
 
-#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
+#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, PartialOrd, Ord)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub enum CurrencyId {
     Native,
```

Comment:
Not sure why we have both `Native` and `StellarNative` here. The native currency of Pendulum is managed by the balances pallet. So we only need lumens here.

Reply:
So, I was researching for a while, and this is a limitation of [orml_currencies](https://docs.rs/orml-currencies/0.4.0/orml_currencies/#currencies-module), we need to specify a `GetNativeCurrencyId` in the configuration of the pallet. The docs says(bold is mine):

>  A native currency ID **could be set** by Config::GetNativeCurrencyId, to identify the native currency.

So for the wording I understand it is optional, but this is actually mandatory in the config, and it needs to be of type `CurrencyId`. So either we keep the `Native` in `CurrencyId`, or we can't use the `orml_currencies` pallet _as it is_.

Reply:
Okay, thanks.